src/components/application_manager/include/application_manager/help_prompt_manager.h

```diff
@@ -0,0 +1,86 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HELP_PROMPT_MANAGER_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HELP_PROMPT_MANAGER_H_
+
+namespace application_manager {
+/**
+ * @brief The HelpPromptManager a mechanism for handling "help" requests
+ * by the user when the application does not provide any such information.
+ */
+class HelpPromptManager {
+ public:
+  /**
+   * @brief Class destructor
+   */
+  virtual ~HelpPromptManager() {}
+  /**
+   * @brief Adds command to constructed values, and send SetGlobalProperties if
+   * required
+   */
+  virtual void OnVrCommandAdded(uint32_t cmd_id,
+                                const smart_objects::SmartObject& command) = 0;
+  /**
+   * @brief Removes command from constructed values, and send
+   * SetGlobalProperties if required
+   */
+  virtual void OnVrCommandDeleted(uint32_t cmd_id) = 0;
+  /**
+   * @brief Send SetGlobalProperties with constructed vrHelp and/or helpPrompt
+   * parameters
+   */
+  virtual void OnTimeoutExpired() = 0;
+  /**
+   * @brief Stop constructing vrHelp and/or helpPrompt if they are present in
+   * message
+   * @param msg containing GlobalProperties
+   * @param is_response determines is the request or response for the for
+   * the SetGlobalPropertiesRequest
+   */
+  virtual void OnSetGlobalPropertiesReceived(
+      const smart_objects::SmartObject& msg, bool is_response) = 0;
+  /**
+   * @brief Starting commands buffering for the requests
```

Comment:
@SNiukalov `Starts`

Reply:
Thanks, fixed in https://github.com/smartdevicelink/sdl_core/pull/2186/commits/44644ccc941b6b188475309630bcce6125e3de84